lazy_tensor_core/lazy_tensor_core/csrc/compiler/backend_impl_interface.h

```diff
@@ -70,22 +71,31 @@ class BackendImplInterface {
    * Device Configuration
    * */
 
-  virtual std::string GetDefaultDevice() const = 0;
+  // Set or get the default device type.
+  // For backends used with virtual c10:: Devices, this configures what real device type
+  // the backend should use, and matters if the backend supports more than one type of real
+  // device.
+  virtual torch_lazy_tensors::BackendDeviceType GetDefaultDeviceType() const = 0;
+  virtual void SetDefaultDeviceType(std::string device_type) const = 0;
```

Comment:
I think 'DefaultDeviceType' might be an unclear name.

What about {Set/Get}VirtualDeviceHardwareType?  

The point is, this DeviceType is more than just a default; it's the type we always use when you use our virtual device.  It's fine that there is a 'default' HardwareType, and that can be returned by GetVirtualDeviceHardwareType by default.

I'm not sure what this implies about using multiple kinds of hardware in the same program.  (A) not sure we want to support this, (B) if we do support it, is it just done by calling Set multiple times in the program?